client/fingerprint/network_unix.go

```diff
@@ -38,10 +39,13 @@ func (f *NetworkFingerprint) Fingerprint(cfg *config.Config, node *structs.Node)
 	if "darwin" == runtime.GOOS {
 		defaultDevice = "en0"
 	}
+	if cfg.Iface != "" {
```

Comment:
Put a comment above to the tune of:

// Allow the user to override the default network device.